[Yang Shiyuan] iP

Timothyoung97 opened the pull request:
# Peter Griffin MBBPH (My Big Boy Peter Helper)

> ┏ (゜ω゜)=☞ 
> 
> Here're some jokes to lighten your mood before you get all serious and read my codes: 😁
> 1. “I have an idea so smart that my head would explode if I even began to know what I was talking about.” — Peter Griffin, Family Guy, Season 2: Love Thy Trophy
> 2. “I can be just as non-competitive as anybody. Matter of fact, I'm the most non-competitive. So I win.” — Peter Griffin, Family Guy, Season 2: Running Mates
> 3. “Now I may be an idiot, but there's one thing I am not sir, and that sir, is an idiot.” — Peter Griffin, Family Guy, Season 5: Hell Comes to Quahog
> 4. *(more? find them yourselves @ [Family Guy](https://youtu.be/ShBlLLsGcXs))!*

## 1. What am I? 😏

```

▒█▀▀█ █▀▀ ▀▀█▀▀ █▀▀ █▀▀█ 　 ▒█▀▀█ █▀▀█ ░▀░ █▀▀ █▀▀ ░▀░ █▀▀▄ 　 ▒█▀▄▀█ ▒█▀▀█ ▒█▀▀█ ▒█▀▀█ ▒█░▒█ 
▒█▄▄█ █▀▀ ░░█░░ █▀▀ █▄▄▀ 　 ▒█░▄▄ █▄▄▀ ▀█▀ █▀▀ █▀▀ ▀█▀ █░░█ 　 ▒█▒█▒█ ▒█▀▀▄ ▒█▀▀▄ ▒█▄▄█ ▒█▀▀█ 
▒█░░░ ▀▀▀ ░░▀░░ ▀▀▀ ▀░▀▀ 　 ▒█▄▄█ ▀░▀▀ ▀▀▀ ▀░░ ▀░░ ▀▀▀ ▀░░▀ 　 ▒█░░▒█ ▒█▄▄█ ▒█▄▄█ ▒█░░░ ▒█░▒█
```

Clearly, I am here to clean up Tim's mess, this is what I do daily for this lazy boy:
- Greet him with some funny words. 👋
- Scold him and make him continue working on his project. 😠
- Keep track of his todos... *(He's from CS, he's quite busy apparently)* 😔
- Remind him of his random events and his crazy deadlines. 😰

## 2. Do you want to get a piece of me? 🎃

### **Here's what you're gonna do!**

1. Download this [`Bot.jar`](https://github.com/Timothyoung97/ip/releases/download/A-Jar/Bot.jar) file.
2. Open up your preferred terminal.
3. Navigate to the directory where you save `Bot.jar`.
4. Run the following command to start using me!
   - `java -jar Bot.jar`
5. Enjoy and have fun!

## 3. Features Development 👩‍💻

- [X] Repond to the following commands:
```json
Command     Custom Input                                                         Purpose
todo        something                                                            -> Add a todo type task into the task list
event       something /at yyyy-mm-ddThh:mm:ss (Format: ISO_LOCAL_DATE_TIME)      -> Add an event type task into the task list
deadline    something /by yyyy-mm-ddThh:mm:ss (Format: ISO_LOCAL_DATE_TIME)      -> Add an deadline type task into the task list
list                                                                             -> Display all tasks with their status
done        Integer (A task indexed in the task list)                            -> Mark a task as completed
delete      Integer (A task indexed in the task list)                            -> Delete a task from the task list
find        keywords (eg find homework play)                                     -> Find tasks that contain the respective keywords
```
- [ ] Accept more date time format
- [ ] Reminders
- [ ] Filter, Sorting

## 4. Make me better? 📈

Prerequisites: JDK 11, update Intellij to the most recent version.

1. Open Intellij (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project first)
2. Open the project into Intellij as follows:
   1. Click `Open`.
   1. Select the project directory, and click `OK`.
   1. If there are any further prompts, accept the defaults.
3. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
   In the same dialog, set the **Project language level** field to the `SDK default` option.
4. After that, locate the `src/main/java/Bot.java` file, right-click it, and choose `Run Bot.main()` (if the code editor is showing compile errors, try restarting the IDE). 

```java
Bot.java

public class Bot {
    public static void main(String[] args) {
        new BotBrain().initiate();
    }
}
```

5. If the setup is correct, you should see something like the below as the output:

```ruby
	                                    ,wpp@@ppwa,
	                                ,p@R@@@@@@@@@MMM@p,
	                              a@@@MMMM%MDZ"|||,p[]g,
	                            ,@K"@@N;j**""`"uww- x   U,
	                           )TK*`""* {; "   [ +*"**,zH;..`>
	                          ,TR|       Yq,,,4-       H    ;  V
	                          BKH       .w;     "KWwwK"`      ; U
	                         ]KR          ``*+<a,JJ,aw~o`      ;"
	                         BKH                       ;;N       U
	                        jTK|                      .A-        {
	                        {KR                      -D;        |j
	                        [KR                        ;`p       J
	                        lKm            ]|            {       J
	                       z@KR            'M.|  ,AQm.|,aH       J
	                      Aj`%K             '*TT"'|`"''          J<
	                 ,~Q;KEMkK"v-                               .C Y.
	             ,wCmKK*"''"KC  `*;                            y" ,U  `>.
	          ,mTmKK'        'Y,   "<;                       ,4  ,-       *.
	        wEmKK"              "y    '*E-                 )*   A           `>
	      mTKKKP                  `*u     "*VL;         ,W`   z"    [          V`y
	   ,Am**"""**<,                   "~,      `"*~v; yPw,  x`      1          { j*
	 ,BjKK*"''      `*,                  `"~,    A   *H  `Y`         @         U A 1
	 2Ek6,gpwwwww,     ".                     '"*    "n               Mm     ,G    C
	[I$mTKKR"|    -Ey    \   /                         "               \K,,aP    ,CE
	`4KKKKR          |\   [z`                           "p              'B;Jw ,mE  H
	  mKKR            ;" 1/                              "p              '@R"'    ]
	  {KKH              \]                              "*?pz-~            @m     H
	   @KH               U                                  y               M|   /
	   `@R              ;`                                   \               H  j
	     VM               $                                   \              J- H
	      \m|              U                                 UUU              H;]
	       B%|      `      [                                   ]              E j
	    ,@R"*KM   lR|      ]                                   J              H CX
	     %@wK&"*@pE$m      [                                   J             ]`j ]
	       `BH iB| ]B`"p   U                                 -,I         z  w;|,A
	        `@mKBpl$[  ( ,B.                                 `,U       ,(g@`
	         1@@M@@@@ppa@@@@pp@Kp.     V.                    z"    ,aw@@@@M
	          %@@@@@M@@@@@@@@@@@@@@@@@ppE*@pp,,.     .,,pww%T%K@@@@@M@@M@@R
	           M@@@@@@@@@MMM@@@@@@@@@@M@@@@@@@@@@@@@@@KKKKKKKKj@@@MMM@@@MM
	            %@@@@@@@@@@@@@@@@@MMMMM@@MM@@@@@@@@@MMMppkpp@@@@@@@@@@@MR
	             "@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@P
	               %@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@R
	                "@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@M*
	                 @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
	                 @@@@@@@@@@@@@@@@@@@@@@@@@@@@@M@@@@@@@@@@@@@@]
	                ]@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@%
	                @@@@@@@@@@@@@@@@@@@@@@@@@@M@@@@@@@@@@@@@@@@@@R
	               ]@@@@@@@@@@@@@@@@@@@@@@@@@MM@@@@@@@@@@@@@@@@@@@U
	               $@@@@@@@@@@@@@@@@@@@@@@@@@MM@@@@@@@@@@@@@@@@@@@@
	               @@@@@@@@@@@@@@@@@@@@@@@@@@@M@@@@@@@@@@@@@@@@@@@R
	               @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@M@@@U
	               8@@@@@@@@@@@@@@@@@@@@@@@@@@$@@@@@@@@@@@@@@@@@@@@@@W
	               1@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@N
	                %@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
	                [@@@@@@@@@@@@@@@@@@@@@@@@@@@@@R@@@@@@@@@@@@@@@@@@@@@M
	                1@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@MM*`
	                  `"%MM@@@@@@@@@@@@@@@@@@@@@MM"   `'"````"'-
	                         """"**PMMMMMP**"`-
	  .--.      .-'.      .--.      .--.      .--.      .--.      .`-.      .--.
	:::::.\::::::::.\::::::::.\::::::::.\::::::::.\::::::::.\::::::::.\::::::::.\
	'      `--'      `.-'      `--'      `--'      `--'      `-.'      `--'      `
	Hallo! My name's Peter!
	How may I be of service to you?
	  .--.      .-'.      .--.      .--.      .--.      .--.      .`-.      .--.
	:::::.\::::::::.\::::::::.\::::::::.\::::::::.\::::::::.\::::::::.\::::::::.\
	'      `--'      `.-'      `--'      `--'      `--'      `-.'      `--'      `
```

Commit:
Add Gradle support

Commit:
Add Gradle support

Timothyoung97 committed:
Level 1 - add greeting

Timothyoung97 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Timothyoung97 committed:
Level 1 - Add Echo & Exit

Timothyoung97 committed:
Level 1 - Add Echo & Exit

Timothyoung97 committed:
Complete Level 2 - Add, List

Timothyoung97 committed:
Complete Level 2 - Add, List

Timothyoung97 committed:
Complete Level 3: Add A Class

Timothyoung97 committed:
Complete Level 3: Add A Class

Timothyoung97 committed:
Complete Level 3: Mark as Done

Timothyoung97 committed:
Complete Level 3: Mark as Done

Timothyoung97 committed:
Complete Level 4

* Make class Task to become abstract class
* Add ToDo, Event, Deadline class. All of them are inherited from Task.

Timothyoung97 committed:
Complete Level 4

* Make class Task to become abstract class
* Add ToDo, Event, Deadline class. All of them are inherited from Task.

Timothyoung97 committed:
Refactor - OOP Principle

Refactor Duke.java to better structure the Bot with OOP principle.

Strip Duke.java into 4 components:
1. Bot.java - imitate a whole bot's body, which currently contains the Bot's Brain
2. BotBrain.java - imitate a bot's brain, which handles the interactions with the user
3. BotPrinter.java - imitate a bot's information output to the user
4. BotMemory.java - imitate a bot's memory unit, where it stores the commonly used lines

Timothyoung97 committed:
Refactor - OOP Principle

Refactor Duke.java to better structure the Bot with OOP principle.

Strip Duke.java into 4 components:
1. Bot.java - imitate a whole bot's body, which currently contains the Bot's Brain
2. BotBrain.java - imitate a bot's brain, which handles the interactions with the user
3. BotPrinter.java - imitate a bot's information output to the user
4. BotMemory.java - imitate a bot's memory unit, where it stores the commonly used lines

Timothyoung97 committed:
Merge branch 'master' of https://github.com/Timothyoung97/ip

Timothyoung97 committed:
Complete A-TextUiTesting

Output with no issue based on the test case provided

Timothyoung97 committed:
Complete A-Exceptions

Add exceptions for:
* invalid commands
* invalid command format
* empty task list
* task out of range

Timothyoung97 committed:
Complete Level 5 - Handle Errors

BotBrain.java
1. change 'if ... else ...' to 'switch' in classifyTask() method
to identify the distinct task

BotMemory.java
1. Added in commonly used error prompting messages

Timothyoung97 committed:
Complete Level 6 - Delete

Add in the function of delete task from task list

Timothyoung97 committed:
Change EXPECT.TXT and input.txt

include some test cases that test the functions of level 5 and 6

Timothyoung97 committed:
Add javaDocs

Timothyoung97 committed:
Amend javaDoc

Timothyoung97 committed:
Complete A-Enums

To make interact() method in BotBrain.java more readable:
* Include enum CommandInput
* Include identifyCommand() method to identify the respective command
* Modify if... else... to switch in interact() method

Timothyoung97 committed:
Refactor identidyCommand() and classifyTask() method in BotBrian.java

Timothyoung97 committed:
Modify Task, Deadline, Event, Todo class

* To support the operation of the formation of task list in a String format

Timothyoung97 committed:
Implement saveToHardDisk() method

Support for save to data.txt
* Refactor switch case as reactToCommand() in BotBrain.java
* Include isTerminated boolean kill switch that ends the program when "bye" is entered

Timothyoung97 committed:
Add Constructor for Task classes

To support feature of loading task from txt files

Timothyoung97 committed:
Complete Level-7

Modification:
* ToDo, Event, Deadline classes -> Add overloading constructors, various get methods
* BotMemory class -> Add save and load features.
* BotException class -> Add new exception type

Timothyoung97 committed:
Modify .gitignore to ignore data.txt

Timothyoung97 committed:
Test for branching features

Timothyoung97 committed:
Merge branch 'test-branching'

* test-branching:
  Test for branching features

Timothyoung97 committed:
Delete try.txt

Timothyoung97 committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Modify .gitignore to ignore data.txt
  Complete Level-7
  Add Constructor for Task classes
  Implement saveToHardDisk() method
  Modify Task, Deadline, Event, Todo class

Timothyoung97 committed:
Revert "Modify .gitignore to ignore data.txt"

This reverts commit dbe7972b046cff14628f052751f07574cfd5f12b.

Timothyoung97 committed:
Implement basic feature of date & time

1) Modify constructors of deadline, event and todo class to accept LocalDate type
2) Modify BotBrain and BotMemroy unit to accept LocalDate type as temporal data
3) Implement basic string to LocalDate conversion in BotTemporal Unit

Timothyoung97 committed:
Extending LocalDate to LocalDateTime

1) Modify Deadline, Event, Task, Todo classes to work with LocalDateTime
2) Modify load/save method in BotMemory class

Timothyoung97 committed:
Complete Level-8

1) Pending Feature:
Decipher different format of temporal data input in string

Timothyoung97 committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Complete Level-8
  Extending LocalDate to LocalDateTime
  Implement basic feature of date & time
  Revert "Modify .gitignore to ignore data.txt"

Timothyoung97 committed:
Complete A-MoreOOP

* Refactorization:
1) BotBrain -> Main Hub that pull in all Units together and react to human response
2) BotCommandResponderUnit -> Decipher command from user and produce the respective action
3) BotDynamicMemoryUnit -> Saving and Loading data.txt
4) BotStaticMemoryUnit -> Serve as a storage space for messages
5) BotTaskStatusGeneratorUnit -> Generate status of tasklist
6) BotTemporalUnit -> Incharge of converting String data to DateTime

Timothyoung97 committed:
Merge branch 'A-MoreOOP'

* A-MoreOOP:
  Complete A-MoreOOP

Timothyoung97 committed:
Packaging of files

Organize files:
1) error -> contains all exceptions classes
2) function -> contains classes that handles UI and feedback to users
3) task -> contains Task type classes
4) memory -> contains classes that handles data

Timothyoung97 committed:
Merge branch 'A-Packages'

* A-Packages:
  Packaging of files

Timothyoung97 committed:
Set save/load file path to relative path

Timothyoung97 committed:
Merge branch 'master' of https://github.com/Timothyoung97/ip into A-JUnit

* 'master' of https://github.com/Timothyoung97/ip:
  Set save/load file path to relative path

Timothyoung97 committed:
Complete A-JUnit (Basic Level)

Test for:
1. Task class
2. ToDo class
3. Event class
4. Deadline class

Timothyoung97 committed:
Merge branch 'A-JUnit'

* A-JUnit:
  Complete A-JUnit (Basic Level)

Timothyoung97 committed:
Complete A-Jar

1) Modify LOGO to non-extendded ACSII text

Timothyoung97 committed:
Merge branch 'A-Jar'

* A-Jar:
  Complete A-Jar

Timothyoung97 committed:
Complete A-JavaDoc

Timothyoung97 committed:
Merge branch 'fake-master' of https://github.com/Timothyoung97/ip into A-CodingStandard

* 'fake-master' of https://github.com/Timothyoung97/ip:
  Complete A-JavaDoc

Timothyoung97 committed:
Complete A-CodingStandard

Timothyoung97 committed:
Merge branch 'A-CodingStandard' of https://github.com/Timothyoung97/ip into fake-master

* 'A-CodingStandard' of https://github.com/Timothyoung97/ip:
  Complete A-CodingStandard

Timothyoung97 committed:
Merge branch 'fake-master' of https://github.com/Timothyoung97/ip into Level-9

* 'fake-master' of https://github.com/Timothyoung97/ip:
  Complete A-CodingStandard
  Complete A-JavaDoc

Timothyoung97 committed:
Complete Level-9 Feature

Add find feature to Bot:
1) find task based on keywords

Follow Up:
1) Refactor
2) JavaDoc for new methods

Timothyoung97 committed:
Merge branch 'fake-master'

* fake-master:
  Complete Level-9 Feature
  Complete A-CodingStandard
  Complete A-JavaDoc

Timothyoung97 committed:
Edit JavaDoc

Timothyoung97 committed:
Merge branch 'branch-A-JavaDoc' of https://github.com/Timothyoung97/ip into branch-A-CodingStandard

* 'branch-A-JavaDoc' of https://github.com/Timothyoung97/ip:
  Edit JavaDoc

Timothyoung97 committed:
Minor edit on java style

Timothyoung97 committed:
Edit on level 9

Timothyoung97 committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Edit on level 9

Timothyoung97 committed:
Merge branch 'branch-A-CodingStandard'

TohWeiJie01 commented on src/main/java/bot/assembly/BotBrain.java:
```diff
@@ -0,0 +1,107 @@
+        CommandInput commandInitial = botCommandResponderUnit.identifyCommand(input);
+
+        switch (commandInitial) {
+            case BYE:
```
Should there be an indentation for the case clauses?

TohWeiJie01 commented on src/main/java/bot/assembly/BotBrain.java:
```diff
@@ -0,0 +1,107 @@
+package bot.assembly;
+
```
Should there be two empty lines here?

TohWeiJie01 commented on src/main/java/bot/assembly/function/BotCommandResponderUnit.java:
```diff
@@ -0,0 +1,317 @@
+
+        switch (taskType) {
+
+            case TODO:
```
Should the case clauses have indentation (similarly for the other switch blocks in your code)?

TohWeiJie01 commented on src/main/java/bot/assembly/function/BotTaskStatusGeneratorUnit.java:
```diff
@@ -0,0 +1,74 @@
+package bot.assembly.function;
+
+import bot.assembly.memory.BotStaticMemoryUnit;
+import bot.assembly.memory.BotDynamicMemoryUnit;
+import bot.assembly.error.EmptyTaskListException;
+
+import bot.assembly.task.Task;
+
+import java.util.List;
```
Should the order and arrangement of the imports be this way (similar to the other imports in your code)?

TohWeiJie01 commented on src/main/java/bot/assembly/function/BotTaskStatusGeneratorUnit.java:
```diff
@@ -0,0 +1,74 @@
+                taskTracker.get(taskTracker.size()-1).toString()
+        );
+
+        output+= String.format(
```
Should there be a whitespace after output?

TohWeiJie01 reviewed (commented):
LGTM. There are only some nix I found and most of the nix are the same. Good job 😄!

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Merge branch 'master' of https://github.com/Timothyoung97/ip

* 'master' of https://github.com/Timothyoung97/ip:
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md

Timothyoung97 committed:
Complete A-Gradle

Timothyoung97 committed:
Merge branch 'branch-Gradle'

* branch-Gradle:
  Complete A-Gradle

Timothyoung97 committed:
Complete A-CheckStyle

Skip CheckStyle for BotStaticMemoryUnit.java

Timothyoung97 committed:
Merge branch 'branch-Checkstyle'

* branch-Checkstyle:
  Complete A-CheckStyle

Timothyoung97 committed:
Complete Javafx tutorial 2

Timothyoung97 committed:
Complete Javafx - Tutorial 3

Timothyoung97 committed:
Complete JavaFX - Tutorial 4

Timothyoung97 committed:
Change profile image

Timothyoung97 committed:
Minor edit on styling

SajalVaishnav commented on src/main/java/bot/assembly/BotBrain.java:
```diff
@@ -0,0 +1,105 @@
+    /**
+     * A method that initiate reaction of Bot to user
+     */
+    private void interact() {
```
This seems to be a grey area between the bot and the UI, maybe more modular code can help make things clearer?

SajalVaishnav commented on src/main/java/bot/assembly/error/EmptyTaskListException.java:
```diff
@@ -0,0 +1,15 @@
+/**
+ * An exception that will be thrown when task list is empty
+ */
+public class EmptyTaskListException extends Exception {
```
Handling exceptions beyond the requirements! Awesome!

SajalVaishnav commented on src/main/java/bot/assembly/function/BotCommandResponderUnit.java:
```diff
@@ -0,0 +1,303 @@
+/**
+ * A class that handles the command input of the user
+ */
+public class BotCommandResponderUnit {
```
This is a huge class that does many things, is it possible to let some of the components handle their own responsibilities rather than having them all be done by this class?

SajalVaishnav commented on src/main/java/bot/assembly/memory/BotStaticMemoryUnit.java:
```diff
@@ -0,0 +1,93 @@
+/**
+ * A class that serves as a storage unit for (final) item
+ */
+public class BotStaticMemoryUnit {
```
B-E-A-U-T-I-F-U-L

SajalVaishnav commented on src/test/java/bot/assembly/task/DeadlineTest.java:
```diff
@@ -0,0 +1,123 @@
+public class DeadlineTest {
+
+    @Test
+    public void testDeadlineToString() {
```
maybe having the input type in the function name can be better?

SajalVaishnav reviewed (commented):
LGTM!

Timothyoung97 committed:
Add message in BotStaticMeoryUnit.java

Timothyoung97 committed:
Implement BotPrinter.java with varargs

Timothyoung97 committed:
Change style dialogBox.java

Timothyoung97 committed:
Edit BotTaskStatusGeneratorUnit.java 's method to return String

Timothyoung97 committed:
Implement basic interaction between bot and user

Follow up:
1) Load / save into hard disk

Timothyoung97 committed:
Complete Level 10

Complete Basic GUI features

Timothyoung97 committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Complete Level 10
  Implement basic interaction between bot and user
  Edit BotTaskStatusGeneratorUnit.java 's method to return String
  Change style dialogBox.java
  Implement BotPrinter.java with varargs
  Add message in BotStaticMeoryUnit.java
  Minor edit on styling
  Change profile image
  Complete JavaFX - Tutorial 4
  Complete Javafx - Tutorial 3
  Complete Javafx tutorial 2

Timothyoung97 committed:
Change Background and send button style

Timothyoung97 committed:
Complete MainWindow GUI

Timothyoung97 committed:
Merge branch 'Improve-GUI'

* Improve-GUI:
  Complete MainWindow GUI
  Change Background and send button style

Timothyoung97 committed:
Configure build.gradle to be compatible with javafx

Update from merged branch (Improve-GUI)
1. Update background of the window
2. Change the send button to Peter icon
3. Change textbox style to machine like terminal
4. Update Peter greeting icon

Follow up:
1. Dialogue box modification

Timothyoung97 committed:
Complete A-Streams

Stream features implemented to improve readability of code

In BotCommandResponderUnit.java, stream is used to add all tasks with the same keyword to a String list for printing.
In BotTaskStatusGeneratorUnit.java, stream is used to convert all tasks in task list to reporting format for reporting.
In BotDynamicMemoryUnit.java. stream is used to convert all tasks in task list to data saving format.

Timothyoung97 committed:
Merge branch 'A-Streams' of https://github.com/Timothyoung97/ip into A-CodeQuality

* 'A-Streams' of https://github.com/Timothyoung97/ip:
  Complete A-Streams

Timothyoung97 committed:
Complete A-Assertions

Implement assertion test in 2 region:

1) BotBrain.java: assert is inserted here to detect fall through error.
If the program is terminated, we should never reach that line of the code.
2) BotCommandResponderUnit.java: assert is inserted here to detect parsing error for
TODO, EVENT, DEADLINE task type. If we ever were to assert this code, it means that
there is some parsing errors to be corrected.

Timothyoung97 committed:
Merge branch 'A-Assertions' of https://github.com/Timothyoung97/ip into A-CodeQuality

* 'A-Assertions' of https://github.com/Timothyoung97/ip:
  Complete A-Assertions

Timothyoung97 committed:
Delete Dead Code (Code Quality)

To prevent confusion and to improve readability.
1. Remove various dead code from CLI implementation of Bot.

Timothyoung97 committed:
Correct variable naming (Code Quality)

Timothyoung97 committed:
Improve Code Quality

1. BotStaticMemoryUnit.java: remove CLI string logo
2. BotDynamicMemoryUnit.java: Indentation
3. BotTemporalUnit.java: Indentation
4. BotTaskStatusGeneratorUnit.java: Remove dead code
5. BotPrinter.java: remove CLI string interface
6. BotCommandResponderUnit.java: Remove dead code and Indentation

Timothyoung97 committed:
Merge pull request #3 from Timothyoung97/A-Streams

Complete A-Streams

Timothyoung97 committed:
Merge pull request #1 from Timothyoung97/A-Assertions

Complete A-Assertions

Timothyoung97 committed:
Merge pull request #2 from Timothyoung97/A-CodeQuality

A code quality

Timothyoung97 committed:
Merge branch 'A-CodeQuality'

* A-CodeQuality:
  Improve Code Quality
  Correct variable naming (Code Quality)
  Delete Dead Code (Code Quality)
  Complete A-Assertions
  Complete A-Streams

Timothyoung97 committed:
Merge branch 'master' of https://github.com/Timothyoung97/ip

* 'master' of https://github.com/Timothyoung97/ip:

Timothyoung97 committed:
Complete A-CI

Timothyoung97 committed:
Revert "Complete A-CI"

This reverts commit 4300120f1548983f67172ea3c6a758c6a3aa8e10.

Timothyoung97 committed:
Create gradle.yml

Timothyoung97 committed:
Complete C-MassOps

Add mass delete & done feature:
command format: massops delete/done

Timothyoung97 committed:
Merge pull request #4 from Timothyoung97/C-MassOps

Complete C-MassOps

Timothyoung97 committed:
Merge branch 'master' of https://github.com/Timothyoung97/ip

* 'master' of https://github.com/Timothyoung97/ip:
  Complete C-MassOps

Timothyoung97 committed:
Code Quality Refactoring

Abstract out the methods which verify the correctness of the command into a new class:
BotCommandChecker.java

Timothyoung97 committed:
Correct logic implementation

EmptyTaskList:
Correct empty task list logic to not throw an exception, because it is not an error for task list to be empty

Timothyoung97 committed:
Complete A-BetterGui

Timothyoung97 committed:
Merge branch 'A-BetterGui'

* A-BetterGui:
  Complete A-BetterGui

Timothyoung97 committed:
Finalize features

Timothyoung97 committed:
Add feature to exit after bye command

Timothyoung97 committed:
Add comments

Timothyoung97 committed:
Add Ui.png

Timothyoung97 committed:
Set theme jekyll-theme-hacker

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Set theme jekyll-theme-hacker

Timothyoung97 committed:
Merge branch 'master' of https://github.com/Timothyoung97/ip

* 'master' of https://github.com/Timothyoung97/ip:
  Set theme jekyll-theme-hacker
  Update README.md
  Set theme jekyll-theme-hacker

Timothyoung97 committed:
Edit code comment and gradle.yml

Timothyoung97 committed:
Revert "Edit code comment and gradle.yml"

This reverts commit c40f66ba4fd4c02f373ef0ab8d459dc0dbea076c.

Timothyoung97 committed:
Update comment

Timothyoung97 committed:
Update gradle.yml

Timothyoung97 committed:
Merge branch 'master' of https://github.com/Timothyoung97/ip

* 'master' of https://github.com/Timothyoung97/ip:
  Update gradle.yml

Timothyoung97 committed:
Remove unused class

Timothyoung97 committed:
Comply with Checkstyle

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Merge branch 'master' of https://github.com/Timothyoung97/ip

* 'master' of https://github.com/Timothyoung97/ip:
  Update README.md
  Update README.md
  Update README.md

Timothyoung97 committed:
Edit Greeting message

Timothyoung97 committed:
change ui.png

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
edit build.gradle

Timothyoung97 committed:
Merge branch 'master' of https://github.com/Timothyoung97/ip

* 'master' of https://github.com/Timothyoung97/ip:
  Update README.md
  Update README.md

Timothyoung97 committed:
Add files via upload

Timothyoung97 committed:
Merge branch 'master' of https://github.com/Timothyoung97/ip

* 'master' of https://github.com/Timothyoung97/ip:
  Add files via upload

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Merge branch 'master' of https://github.com/Timothyoung97/ip

* 'master' of https://github.com/Timothyoung97/ip:
  Update README.md
  Update README.md

Timothyoung97 committed:
Modify build.gradle

Timothyoung97 committed:
Change on fxml's javafx version from 16 to 11

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Update README.md

Timothyoung97 committed:
Temp change

Timothyoung97 committed:
Edit grammar
